src/query.rs

```diff
@@ -234,6 +239,96 @@ impl From<Vec<Arc<OverrideMap>>> for OverrideStack {
     }
 }
 
+/// Data for generating a **witness** from the results of a [`SemverQuery`].
+///
+/// A witness is a minimal compilable example of how downstream code would
+/// break given this change.  See field documentation for more information
+/// on each member.
+///
+/// Fields besides [`hint_template`](Self::hint_template) are optional, as it is not
+/// always necessary to use an additional query [`witness_query`](Self::witness_query)
+/// or possible to build a compilable witness from [`witness_template`](Self::witness_template)
+/// for a given `SemverQuery`.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Witness {
+    /// A [`handlebars`] template that renders a user-facing hint to give a quick
+    /// explanation of breakage.  This may not be a buildable example, but it should
+    /// show the idea of why downstream code could break.  It will be provided any
+    /// `@output` query declarations in the [`SemverQuery`] query.
```

Comment:
```suggestion
    /// show the idea of why downstream code could break.  It will be provided all
    /// `@output` data from the [`SemverQuery`] query that contains this [`Witness`].
```